edgedb-protocol/src/codec.rs

```diff
@@ -525,7 +525,7 @@ fn decode_input_tuple<'t>(mut elements:DecodeInputTuple, codecs:&Vec<Arc<dyn Cod
 fn decode_tuple<'t>(mut elements:DecodeTupleLike, codecs:&Vec<Arc<dyn Codec>>) -> Result<Vec<Value>, DecodeError>{
     codecs
         .iter()
-        .map(|codec| codec.decode(required_element(elements.read()?)?))
+        .map(|codec| codec.decode(elements.read()?.ok_or_else(|| errors::MissingRequiredElement.fail::<()>().err().unwrap())?))
```

Comment:
I originally had this, but it was so ugly that I decided to introduce the helper function (plus this check is done in two places)

Reply:
There is a [`.build()` method](https://docs.rs/snafu/0.6.8/snafu/guide/examples/basic/struct.Leaf.html#method.build):
```
codec.decode(elements.read()?.ok_or_else(|| errors::MissingRequiredElement.build())?)
```

Reply:
Thanks for the tip. Using the `build` method this is clean enough. Pushed a new version where I replaced all those places.